### PR TITLE
Get target dirs for Java binaries

### PR DESCRIPTION
**Pull request by eemcmullan**
## Summary by CodeRabbit

- Refactor
  - Streamlined analysis provider initialization for faster, more reliable startup.
- Bug Fixes
  - Honors HTTP/HTTPS proxy settings during provider setup.
  - Avoids unnecessary waiting when target directories already exist.
  - Clearer, more actionable error messages during initialization.
- Tests
  - Added comprehensive tests for provider config generation and Java target path discovery, including edge cases and integration-style scenarios.
- Chores
  - Promoted testing library to a direct dependency for improved test stability.

### CODE REVIEW

**Review by jmle — CHANGES_REQUESTED** [review]
Just left a comment if possible

File: cmd/analyze-bin.go [hunk 1]
```diff
@@ -525,34 +538,131 @@ func (a *analyzeCommand) setConfigsContainerless(configs []provider.Config) []pr
...
+	prov, err := lib.GetProviderClient(config, analysisLog)
+	if err != nil {
+		a.log.Error(err, "failed to create builtin provider")
+		return nil, err
+	}
+
+	return prov, nil
+}
+
+func (a *analyzeCommand) setJavaProvider(config provider.Config, analysisLog logr.Logger) provider.InternalProviderClient {
```
[the remark this answers is not in the document]
Comment (aufi) [reply cmd/analyze-bin.go:564]: I agree this would be cleaner, but so far only kantra container-less analysis was for java-only, so there might come some refactoring in future (bringing more analyzer-like structure for analyze-bin), but from my point of view, this is not merge blocker.

**Review by aufi — APPROVED** [review]

**Review by jmle — APPROVED** [review]
